frontend/public/components/image-stream.tsx

```diff
@@ -30,10 +31,33 @@ export const getBuilderTags = (imageStream: K8sResourceKind): any[] => {
   return _.filter(imageStream.spec.tags, tag => isBuilderTag(tag) && statusTags[tag.name]);
 };
 
+// Sort tags in reverse order by semver, falling back to a string comparison if not a valid version.
+export const getBuilderTagsSortedByVersion = (imageStream: K8sResourceKind): any[] => {
+  return getBuilderTags(imageStream).sort(({name: a}, {name: b}) => {
+    const v1 = semver.coerce(a);
+    const v2 = semver.coerce(b);
+    if (!v1 && !v2) {
+      return a.localeCompare(b);
+    }
+    if (!v1) {
+      return 1;
+    }
+    if (!v2) {
+      return -1;
+    }
+    return semver.rcompare(v1, v2);
+  });
+};
+
+export const getMostRecentBuilderTag = (imageStream: K8sResourceKind) => {
+  const tags: any[] = getBuilderTagsSortedByVersion(imageStream);
```

Comment:
You shouldn't need to declare `: any[]` here since it's inferred from function return type.